### PR TITLE
Cron job

### PR DESCRIPTION
**Pull request by developer-aditya**
Packet AutoSync and AutoUpload [RCF-188]

[RCF-188]: https://mosip.atlassian.net/browse/RCF-188?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pubspec.lock [hunk 1]
```diff
@@ -1315,5 +1307,5 @@ packages:
     source: hosted
     version: "3.1.2"
 sdks:
-  dart: ">=3.1.0 <4.0.0"
-  flutter: ">=3.13.0"
```
Comment (jainhitesh9998) [inline pubspec.lock:1319]: please revert this version change

File: android/app/src/main/java/io/mosip/registration_client/MainActivity.java [hunk 2]
```diff
@@ -147,94 +153,131 @@ public class MainActivity extends FlutterActivity {
...
-
-            registrationList = packetService.getRegistrationsByStatus("CREATED");
-            registrationList.forEach(value->{
+            Log.d(getClass().getSimpleName(), "Sync Packets in main activity");
+            Integer batchSize = getBatchSize();
+            List<Registration> registrationList = packetService.getRegistrationsByStatus(PacketClientStatus.APPROVED.name(), batchSize);
+            for (Registration value : registrationList) {
                 try {
-                    Log.d(getClass().getSimpleName(), "Syncing " + value.getPacketId());
+                    Log.e(getClass().getSimpleName(), "Syncing " + value.getPacketId());
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/MainActivity.java:236]: log.e should only be used where there's error being logged, use log.i or log.d for info or debug log

File: android/app/src/main/java/io/mosip/registration_client/MainActivity.java [hunk 3]
```diff
@@ -147,94 +153,131 @@ public class MainActivity extends FlutterActivity {
...
-            registrationList.forEach(value->{
+    private void uploadRegistrationPackets(Context context) {
+        if(NetworkUtils.isNetworkConnected(context)){
+            Log.d(getClass().getSimpleName(), "Upload Packets in main activity");
+            Integer batchSize = getBatchSize();
+            List<Registration>  registrationList = packetService.getRegistrationsByStatus(PacketClientStatus.SYNCED.name(), batchSize);
+            for (Registration value : registrationList) {
                 try {
-                    Log.d(getClass().getSimpleName(), "Uploading " + value.getPacketId());
+                    Log.e(getClass().getSimpleName(), "Uploading " + value.getPacketId());
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/MainActivity.java:252]: log.e should only be used where there's error being logged, use log.i or log.d for info or debug log

File: android/app/src/main/java/io/mosip/registration_client/MainActivity.java [hunk 4]
```diff
@@ -147,94 +153,131 @@ public class MainActivity extends FlutterActivity {
...
         }
+        return 4;
+    }
+
+    private long getIntervalMillis(String api){
+        AtomicLong alarmTime = new AtomicLong(System.currentTimeMillis()+60000);
+        List<SyncJobDef> syncJobs = syncJobDefRepository.getAllSyncJobDefList();
+        for (SyncJobDef value : syncJobs) {
+            if (Objects.equals(value.getApiName(), api)) {
+                Log.e(getClass().getSimpleName(), String.valueOf(value.getSyncFreq()) + " Cron Expression");
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/MainActivity.java:276]: log.e should only be used where there's error being logged, use log.i or log.d for info or debug log

**Review by jainhitesh9998 — CHANGES_REQUESTED** [review]

File: pubspec.lock (hunk 1 above)
Comment (Piyush7034) [reply pubspec.lock]: Changed on upgrading the dependencies in pubspec.yaml file.

File: android/app/src/main/java/io/mosip/registration_client/MainActivity.java (hunk 2 above)
Comment (Piyush7034) [reply android/app/src/main/java/io/mosip/registration_client/MainActivity.java]: Changed to log.d

File: android/app/src/main/java/io/mosip/registration_client/MainActivity.java (hunk 3 above)
Comment (Piyush7034) [reply android/app/src/main/java/io/mosip/registration_client/MainActivity.java]: Changed to log.d

File: android/app/src/main/java/io/mosip/registration_client/MainActivity.java (hunk 4 above)
Comment (Piyush7034) [reply android/app/src/main/java/io/mosip/registration_client/MainActivity.java]: Changed to log.d

**Pull request merged**